cpu/stm32/include/fdcandev_stm32.h

```diff
@@ -0,0 +1,325 @@
+/** @] */
+
+/**
+ * @name    Birates
+ * @{
+ */
+#ifndef FDCANDEV_STM32_DEFAULT_BITRATE
+#define FDCANDEV_STM32_DEFAULT_BITRATE 500000U /**< Default bitrate for headers and non-FDCAN messages */
+#endif
+#ifndef FDCANDEV_STM32_DEFAULT_FD_DATA_BITRATE
+#define FDCANDEV_STM32_DEFAULT_FD_DATA_BITRATE 1000000U /**< Default FDCAN data bitrate */
+#endif
+/** @} */
+
+#ifndef FDCANDEV_STM32_DEFAULT_SPT
+/** Default sampling-point */
+#define FDCANDEV_STM32_DEFAULT_SPT     875
+#endif
+
+/** bxCAN device configuration */
+typedef struct {
+    FDCAN_GlobalTypeDef *can;   /**< CAN device */
+    uint32_t rcc_mask;          /**< RCC mask to enable clock */
+    gpio_t rx_pin;              /**< RX pin */
+    gpio_t tx_pin;              /**< TX pin */
+    gpio_af_t af;               /**< Alternate pin function to use */
+    bool en_deep_sleep_wake_up; /**< Enable deep-sleep wake-up interrupt */
+    uint8_t it0_irqn;                   /**< Interrupt line 0 IRQ channel */
+    uint8_t it1_irqn;                   /**< Interrupt line 1 IRQ channel */
+    uint8_t ttcm : 1;                   /**< Time triggered communication mode */
+    uint8_t abom : 1;                   /**< Automatic bus-off management */
+    uint8_t awum : 1;                   /**< Automatic wakeup mode */
+    uint8_t nart : 1;                   /**< No automatic retransmission */
+    uint8_t rflm : 1;                   /**< Receive FIFO locked mode */
+    uint8_t txfp : 1;                   /**< Transmit FIFO priority */
+    uint8_t lbkm : 1;                   /**< Loopback mode */
+    uint8_t silm : 1;                   /**< Silent mode */
+} can_conf_t;
+/** can_conf_t is re-defined */
+#define HAVE_CAN_CONF_T
+
+/**
+ * @name    STM32 mailboxes
+ * @{
+ */
+#define FDCAN_STM32_TX_MAILBOXES 3      /**< Number of frame the driver can transmit simultaneously */
+#define FDCAN_STM32_RX_MAILBOXES \
+        (FDCANDEV_STM32_CHAN_NUMOF * 6) /**< Maximum number of frame the driver can receive simultaneously. \
+                                        There are 3 buffers per FIFO and 2 FIFO per channel. */
+/** @} */
+
+
+/** FDCAN candev descriptor */
+typedef struct can can_t;
+/** can_t is re-defined */
+#define HAVE_CAN_T
+
+#define FDCAN_SRAM_MESSAGE_RAM_SIZE     0x350   /**< FDCAN SRAM message size */
+
+/**
+ * @name Message RAM addresses - 32 bits aligned
+ * @{
+ */
+#define FDCAN_SRAM_FLESA                0x1CU   /**< Filter List Extended Start Address */
+#define FDCAN_SRAM_F0SA                 0x2CU   /**< Rx FIFO0 Start Address */
+#define FDCAN_SRAM_F1SA                 0x62U   /**< Rx FIFO1 Start Address */
+#define FDCAN_SRAM_EFSA                 0x98U   /**< Event FIFO Start Address */
+#define FDCAN_SRAM_TBSA                 0x9EU   /**< Tx Buffer Start Address */
+/** @} */
+
+/**
+ * @name Standard filter bit definition
+ * @{
+ */
+#define FDCAN_SRAM_FLS_SFID1_Pos        (16U)                               /**< Standard filter ID 1 position */
```

Comment:
if possible  please keep lines shorter than 100 chars  (the line length warnings  are annoying)

Reply:
done